src/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp

```diff
@@ -329,10 +329,14 @@ static size_t ReadBody(char* ptr, size_t size, size_t nmemb, void* userdata, boo
             {
                 Aws::StringStream chunkedTrailer;
                 chunkedTrailer << "0\r\n";
-                if (request->GetRequestHash().second != nullptr)
-                {
+                if (request->GetRequestHash().second->isPrecalculatedHashSet()) {
+                    chunkedTrailer << "x-amz-checksum-" << request->GetRequestHash().first << ":"
+                                   << request->GetRequestHash().second->GetPrecalculatedHash() << "\r\n";
+                } else if (request->GetRequestHash().second != nullptr) {
+                    auto hash = request->GetRequestHash().second->GetHash().GetResult();
+                    std::cout << hash.GetUnderlyingData();
```

Comment:
looks like some debugging code slipped in here, should remove this and potentially inline the hash variable